core_types/src/eav.rs

```diff
@@ -28,6 +31,9 @@ pub type Attribute = String;
 /// Address of AddressableContent representing the EAV value
 pub type Value = Address;
 
+#[derive(PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord)]
+pub struct Key(pub i64, pub Action, pub Uuid);
```

Comment:
Can we get a rustdoc on this central struct?

I am wondering why we need this complex key, and why a simple unsigned integer for the index is not enough?

Reply:
If you wanted to get a record of the eav at any point in time, you probably are not able to do that with the index only. Also, if we are looking at an agent centric POV, we have to look at it from the point that collisions are handled well. If the UUID is put there it wont overide a time + action enum